### DISCUSSION TITLE
Adding events to WebSocketStream

### DISCUSSION THREAD

**Discussion by tavithedev** [denoland/deno#15628] [Ideas]
It would be great for you to add events to WebSocketStream, as it would make my work lot easier

**Comment by crowlKats** [comment]
I am not sure what you are asking for, could you specify?

**Reply by tavithedev** [reply]
`onopen`, `onclose`, `onerror`, etc...

**Reply by crowlKats** [reply]
use `WebSocket` for that. `WebSocketStream` is meant to be stream based, not event based, as the name suggests. Also, it's not up to us to add functionality to these classes, as we follow specifications.

**Reply by tavithedev** [reply]
websocket doesn't support headers on handshake

**Comment by etodanik** [comment]
This is the right repo to request changes to the spec:
https://github.com/ricea/websocketstream-explainer
